src/stim/diagram/detector_slice/detector_slice_set.cc

```diff
@@ -615,31 +619,37 @@ void DetectorSliceSet::write_svg_diagram_to(std::ostream &out) const {
 
     for (size_t k = 0; k < num_ticks; k++) {
         for (auto q : used_qubits()) {
-            auto c = coords(min_tick + k, q);
+            auto t = min_tick + k;
+            auto qc = coordsys.qubit_coords[q]; // the raw qubit coordinates
+            auto sc = coords(t, q); // the svg coordinates, offset to the correct slice plot
             out << "<circle";
-            write_key_val(out, "cx", c.xyz[0]);
-            write_key_val(out, "cy", c.xyz[1]);
+            write_key_val(out, "id", std::format("qubitdot_{}_{}_{}", qc.xyz[0], qc.xyz[1], t))
```

Comment:
Raw qubit coordinates aren't guaranteed to be 2 dimensional. Include a test case including a circuit with a max of qubit coordinate data, including a qubit with no coordinates, a qubit with 1d coordinates, and a qubit with 3d coordinates.

Reply:
`coords` above also assumes the raw coordinates has 2 elements, does that also need a check?
```
auto result = coordsys.qubit_coords[qubit];
...
result.xyz[0] += coordsys.size.xyz[0] * sx * SLICE_WINDOW_GAP;
result.xyz[1] += coordsys.size.xyz[1] * sy * SLICE_WINDOW_GAP;
return result;
```
More generally, how does the diagramming code even handle >2D qubit coords..? I figure this is just not respected in the current version.

Reply:
The diagramming code uses an orthographic projection to turn the raw coordinate data into 2d points for the SVG canvas (or 3d in some cases if output to a model).